examples/contour/01-crds.yaml

```diff
@@ -6025,7 +6030,12 @@ spec:
                           type: integer
                         mirror:
                           description: If Mirror is true the Service will receive
-                            a read only mirror of the traffic for this route.
+                            a read only mirror of the traffic for this route. If Mirror
+                            is true, then fractional mirroring is enabled by setting
+                            the Weight field. Omitting the Weight field or setting
+                            it to 0 will result in 100% mirroring. Setting it to an
```

Comment:
shouldn't we set it to 100 to send 100% traffic ?

Reply:
In that case, can you explicitly mention the edge case. Is there no way around the 0 value, just that the difference between a value of 0 and 1 is drastic and opposite, can we denote omission of the field by Null value instead of 0, or maybe not allowing a 0 value altogether. I can't think of a case where you will mention mirroring with 0 weights.

Reply:
I do agree; it does tempt the principle of least astonishment.

Being more explicit about that edge is a good idea regardless, ie. stating that legal values are 1-100.

I don't see an easy workaround otherwise, this overloading of the `weight` field was always a bit of a compromise:
- there isn't a non-0 null type for the go int/float; 
- changing the field type (and to what?) would break backwards-compat; 
- adding a new field explicitly, ex. `mirroring_fraction`, seemed unnecessarily complicating;

Have updated the comments in the docs & code comments. Lmk.

Reply:
comment looks perfect! allowing values from 1-100 make sense! go for the merge 🎉